lib/heap/heap_tree.ml

```diff
@@ -15,11 +17,11 @@ module M : Heap_intf.M with type vt = Term.t = struct
     match block with
     | Leaf ((l, r), v) ->
       fprintf fmt "{ \"leaf\": { \"range\": \"[%a, %a]\", \"value\": \"%a\"} }"
-        Term.pp l Term.pp r Term.pp v
+        Expr.pp l Expr.pp r Expr.pp v
     | Node ((l, r), ch) ->
       fprintf fmt
-        "{ \"node\": { \"range\": \"[%a, %a]\", \"children\": [ %a ]} }" Term.pp
-        l Term.pp r
+        "{ \"node\": { \"range\": \"[%a, %a]\", \"children\": [ %a ]} }" Expr.pp
+        l Expr.pp r
```

Comment:
Maybe we can use the string delimiters `{|` and `|}` instead of `"` to avoid having to much escapes here:

```suggestion
        {|{ "node": { "range": "[%a, %a]", "children": [ %a ]} }|} Expr.pp
        l Expr.pp r
```